src/parameter_validation.h

```diff
@@ -71,13 +72,22 @@ namespace Image_Function
             throw imageException( "Bad input parameters in image function" );
     }
 
-    template <typename TImage>
-    void ParameterValidation( const TImage & image1, const TImage & image2, const TImage & image3 )
+    template <typename TImage, typename... Args>
+    void ParameterValidation( const TImage & image1, const TImage & image2, Args... args )
     {
-        if( image1.empty() || image2.empty() || image3.empty() || !IsCorrectColorCount( image1 ) || !IsCorrectColorCount( image2 ) ||
-            !IsCorrectColorCount( image3 ) || image1.width() != image2.width() || image1.height() != image2.height() ||
-            image1.width() != image3.width() || image1.height() != image3.height() )
-            throw imageException( "Bad input parameters in image function" );
+        ParameterValidation( image1, image2 );
+        ParameterValidation( image2, args... );
+    }
+
+    inline std::pair<uint32_t, uint32_t> ExtractRoiSize( uint32_t width, uint32_t height )
```

Comment:
To avoid inline and cross file compilation let's change this function into template function something like this:
```cpp
template <typename _Type>
std::pair<_Type, _Type> ExtractRoiSize( _Type width, _Type height )
{
    return std::pair<_Type, _Type>( width, height );
}
```

Reply:
Sure! I was not sure if having such a generic function would be a good idea.. But that certainly makes life a lit easier